### ISSUE TITLE
Performance testing

### ISSUE THREAD

**Issue by sohrab-** [labeleven-dev/bettercallsol#42]
We need to do some react performance testing to ensure there are no unneeded re-renders, calls to RPC nodes, etc. due to code bugs

**Comment by josephlabeleven** [comment]
We can use React dev tools a Chrome extension (https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi) to see what components are getting re-rendered during any component changes.

![Screen Shot 2022-08-22 at 2 22 12 pm](https://user-images.githubusercontent.com/111264560/185839096-7dde7672-389b-4588-bf1a-0b9a143c5d5b.png)

You will have two more tabs - Components and Profiler and click on the "Settings" to enable Highlight updates when components render.

![Screen Shot 2022-08-22 at 2 23 44 pm](https://user-images.githubusercontent.com/111264560/185839201-051c007a-51e1-47a4-b62a-a909e39a2b04.png)

**Comment by josephlabeleven** [comment]
A performance issue was found that on every changes of data in transaction struct, the whole Transaction component will re-reder, which is causing redundant rerendering.


https://user-images.githubusercontent.com/111264560/185839855-8c63eb33-84ed-4e8f-a64d-9ee7ef71c4cf.mp4


Root cause is found due to the following code as an example:
```
export const Transaction: React.FC = () => {
  const [transaction, set] = useSessionStoreWithUndo((state) => [
    state.transaction,
    state.set,
  ]);

```

Under Transaction.tsx, it gets the transaction struct from the state and pass it down to its children components as prop, all the children components with the Transaction component get updated on every changes within the transaction struct.

Solution would be instead of gettting the transaction struct in the parent component and pass it down to its children compoent, we get the relevant transaction data within the children component.

**Comment by josephlabeleven** [comment]
resolved by #66
